sktime/forecasting/trend/_spline_trend_forecaster.py

```diff
@@ -0,0 +1,215 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements SplineTrendForecaster."""
+
+__author__ = ["tensorflow-as-tf", "mloning", "aiwalter", "fkiraly", "jgyasu"]
+__all__ = ["SplineTrendForecaster"]
+
+import numpy as np
+import pandas as pd
+from sklearn.base import clone
+from sklearn.linear_model import LinearRegression
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import SplineTransformer
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.trend._util import _get_X_numpy_int_from_pandas
+
+
+class SplineTrendForecaster(BaseForecaster):
+    """Forecast time series data with a spline trend."""
```

Comment:
Can you please update this docstring with `__init__` arguments, examples, references, etc.?